### ISSUE TITLE
resample_data is restricted to within +/- 1 sigma when creating the sampled datasets

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#779]
This is something I noticed when looking at #740 

The `resample_data` code only ever creates a value that is within the one-sigma limits for that point. We can see this by adding code to the `ReSampleData.call` method so that it returns the data values that it creates. For this ticket I added the following so that the result contains the `yvals` field (this is the end of `sherpa/sim/__init__.py`):

```python
        rys = []  # ADDED
        numpy.random.seed(seed)
        for j in range(niter):
            ry = []
            for i in range(len(y_l)):
                a = y_l[i]
                b = y_h[i]
                r = -1
                while r < a or r > b:
                    sigma = b - y[i]
                    u = numpy.random.random_sample()
                    if u < 0.5:
                        sigma=y[i]-a
                    r = numpy.random.normal(loc=y[i],scale=sigma,size=None)
                    if u < 0.5 and r > y[i]:
                        r = -1
                    if u > 0.5 and r < y[i]:
                        r = -1
                ry.append(r)

            ry = numpy.asarray(ry)
            rys.append(ry)  # ADDED

            # fit is performed for each simulated data point
            fit = Fit(Data1D('tmp', x, ry), self.model, LeastSq( ), LevMar())
            fit_result = fit.fit()

            for index, val in enumerate(fit_result.parvals):
                name = pars_index[index]
                pars[name].append(val)

        result = {}
        for index, name in pars_index.items():
            avg = numpy.average(pars[name])
            std = numpy.std(pars[name])
            print(name, ': avg =', avg, ', std =', std)
            result[name] = pars[name]

        result['yvals'] = numpy.asarray(rys)  # ADDED
        return result
```

I then have a simple script with a very-simple dataset with asymmetric errors:

```python
import numpy as np

from matplotlib import pyplot as plt

from sherpa.data import Data1DAsymmetricErrs
from sherpa.models.basic import Const1D
from sherpa.sim import ReSampleData

mdl = Const1D()
mdl.c0 = 10

x = np.asarray([1, 3, 5, 7, 9])
y = np.asarray([11, 12, 9, 9, 13])
dy = np.asarray([2, 4, 10, 1, 10])

d = Data1DAsymmetricErrs('test', x, y, dy, dy / 2)

sim = ReSampleData(d, mdl)

res = sim(seed=7329, niter=1000)
ys = res['yvals']

plt.clf()
for i in range(5):
    lbl = 'y={} -{} +{}'.format(y[i], dy[i], dy[i]/2)
    plt.hist(ys[:, i], alpha=0.5, label=lbl)

plt.xlim(-2, 19)
plt.ylim(0, 200)
plt.gca().set_xticks(range(0, 22, 2))
plt.xlabel('Dependent variable')
plt.legend()
plt.savefig('yvals.png')
print("Created: yvals.png")
```

Running this creates the following histograms, showing the distribution of values for each bin, and the labels give the y value and asymmetric errors for the bin from which the values were drawn.

![yvals](https://user-images.githubusercontent.com/224638/79275243-d1895a80-7e73-11ea-872d-a1928adeb200.png)

You can see 

 - the need for #740 - where the range for bin 2 (green) is y=9 -10 + 5; since the one-sigma range is -1 to 14 and so includes -1 all simulated values for this bin are set to 1 (and this is what #740 fixes)

 - for the other bins we can see that the histogram only contains values over the one-sigma range - e.g. bin 4 (purple) us y=13 -10 + 5, and the values are limited to 3 to 18

I expected these histograms to extend beyond the one sigma limits: as there's 1000 iterations then we should see ~ 300 values per bin outside the 1-sigma range.

I'm also a bit concerned that these distributions are not peaking at the "data value" - e.g. the purple distribution is for y=13 -10 + 5 but it peaks at ~15-16. I haven't taken care in creating these histograms, so this needs further investigation. Actually, I guess in this case the upper limits are smaller than the lower limits (by construction), and so this will bias the distributions (e.g. thinking about adding together two halves of a normal, with sigma and sigma/2 for the left and right sides, and then binning them at a scale close to sigma / 2). The other bins (apart from the green one) look better (but that may be in part because the distributions are too small to easily inspect in this plot).

**Comment by DougBurke** [comment]
For reference, here's the graph after applying #740 - so you can see the -1 case has gone, but values are still limited to +/1 sigma

![yvals](https://user-images.githubusercontent.com/224638/79338432-4badf380-7ef5-11ea-8400-7c36fce922f4.png)

**Comment by DougBurke** [comment]
If I add a simple fix to #740 to not cut off at +-1 sigma then the plot looks like

![yvals](https://user-images.githubusercontent.com/224638/79378111-c3e2dc00-7f2a-11ea-8b0c-349bcb7e5778.png)
